src/mono/System.Private.CoreLib/src/System/Threading/LowLevelLifoAsyncWaitSemaphore.Browser.Threads.Mono.cs

```diff
@@ -112,10 +112,14 @@ private void PrepareAsyncWaitForSignal(int timeoutMs, Action<LowLevelLifoAsyncWa
 
         _onWait();
 
+        if (timeoutMs == 0) {
```

Comment:
It looks like `InternalAsyncWaitSuccess` calls `PrepareAsyncWaitCore` and not this method, so maybe the timeout check should be in `PrepareAsyncWaitCore`. There is an assert in this method above that the timeout here shouldn't be 0.